packages/core/src/modules/connections/ConnectionsApi.ts

```diff
@@ -227,6 +232,37 @@ export class ConnectionsApi {
     return connectionRecord
   }
 
+  /**
+   * Send a trust ping to an established connection
+   *
+   * @param connectionId the id of the connection for which to accept the response
+   * @param responseRequested do we want a response to our ping
+   * @param withReturnRouting do we want a response at the time of posting
+   * @returns boolean indicating success or failure
+   */
+  public async sendPing(
+    connectionId: string,
+    { responseRequested = true, withReturnRouting = false }: SendPingOptions
+  ) {
+    const connection = await this.getById(connectionId)
+
+    const { message } = await this.connectionService.createTrustPing(this.agentContext, connection, {
+      responseRequested: responseRequested,
+    })
+
+    // Disable return routing as we don't want to receive a response for this message over the same channel
+    // This has led to long timeouts as not all clients actually close an http socket if there is no response message
+    if (!withReturnRouting) {
+      message.setReturnRouting(ReturnRouteTypes.none)
+    }
```

Comment:
If return routing is true, we should explicitly enable It I think. Otherwise it can lead to unexpected behaviour (I pass `withReturnRouting: true`, but it won't actually use return routing because we have an endpoint.

I would make the default of `withReturnRouting` undefined, then we have three options:
- `withReturnRouting: undefined` -- we don't set the return routing
- `withReturnRouting: true` -- we set the return routing to `ReturnRouteTypes.All`
- `withReturnRouting: false` -- we set the return routing to `ReturnRoutingTypes.None`

Another option would be to make the parameter `returnRouteType` and actually allow the user to pass thread / none / all

Reply:
Done